persist/fs/read.go

```diff
@@ -302,7 +437,20 @@ func (r *reader) entryID(id []byte) ts.ID {
 // NB(xichen): Validate should be called after all data are read because the
 // digest is calculated for the entire data file.
 func (r *reader) Validate() error {
-	return r.dataFdWithDigest.Validate(r.expectedDataDigest)
+	err := r.indexReader.readerWithDigest.Validate(r.expectedIndexDigest)
+	if err != nil {
+		return fmt.Errorf("could not validate index file: %v", err)
+	}
+	// TODO(r): Make sure to validate checksum when using seeker now that
```

Comment:
Is the TODO still applicable?

Reply:
It was and I have implemented that now, thanks for the catch =]